### ISSUE TITLE
Remove dependency on external site cdn.mathjax.org

### ISSUE THREAD

**Issue by raxraxraxraxrax** [probcomp/bayeslite#354]
When cdn.mathjax.org is down, we get a scary-looking warning page upon loading a notebook. It doesn't break core functionality, but it could stop some things from working and we should arguably bring the library in locally so that we don't have to worry about it.

**Comment by raxraxraxraxrax** [comment]
This isn't just a hypothetical; it's down the day before a class session and while we expect it back up we have no control over it. :)

**Comment by riastradh-probcomp** [comment]
I grepped the bayeslite repository for the string 'mathjax'.  I didn't turn anything up.  Where does this come from?

**Comment by riastradh-probcomp** [comment]
I grepped the bdbcontrib repository too -- nothing there either.

**Comment by raxraxraxraxrax** [comment]
I think it might come from ipython itself? Full error message text: 

Failed to retrieve MathJax from 'https://cdn.mathjax.org/mathjax/latest/MathJax.js'

Math/LaTeX rendering will be disabled.

If you have administrative access to the notebook server and a working internet connection, you can install a local copy of MathJax for offline use with the following command on the server at a Python or IPython prompt:

>>> from IPython.external import mathjax; mathjax.install_mathjax()

This will try to install MathJax into the IPython source directory.

If IPython is installed to a location that requires administrative privileges to write, you will need to make this call as an administrator, via 'sudo'.

When you start the notebook server, you can instruct it to disable MathJax support altogether:

$ ipython notebook --no-mathjax

which will prevent this dialog from appearing.

**Comment by riastradh-probcomp** [comment]
Great.  So running ipython even on a local site automatically downloads code from the internet to be executed in your browser?  I suggest we apply --no-mathjax pronto and maybe check for any other such back doors in ipython.

**Comment by raxraxraxraxrax** [comment]
> and maybe check for any other such back doors in ipython.

We'd get this as part of #355 .

**Comment by riastradh-probcomp** [comment]
Not necessarily.  Such back doors are likely to be silent if they temporarily fail.

**Comment by gregory-marton** [comment]
Disagree. People download and use javascript all the time from everywhere as a standard part of their daily internet usage, and web browsers are designed to defend against attacks by malicious javascript. Doing something special ourselves, that other ipython users don't do, seems like serious overkill, and antithetical to our mission to spend our time advancing the state of the art in probabilistic computing, rather than in internet security.
On the other hand, making our stuff usable offline seems like a big win for the user's convenience, and ultimately for their security, if they wish to audit and isolate the systems they use.

**Comment by riastradh-probcomp** [comment]
I expect scripts that run from local files to be more dangerous than ones that run from remote hosts because of the same-origin policy: scripts that run from local files have access to your file system, whereas those from remote hosts do not.

I agree that we shouldn't have to spend time on this.  That's why I systematically encourage avoiding these questions by not introducing back doors in our own software and by rejecting external software that uses them.

**Comment by raxraxraxraxrax** [comment]
So, the bayesdb on my local machine _still_ can’t load MathJax. I’m beginning to wonder if it is something to do with my machine; I’ve tried multiple versions of bayesdb. It’s _especially_ weird because I can access the file https://cdn.mathjax.org/mathjax/latest/MathJax.js just fine from a web browser. We should arguably just distribute a copy of this, MathJax is important.

**Comment by riastradh-probcomp** [comment]
@raxraxraxraxrax Can you find anything in your browser's console relevant to it?

**Comment by riastradh-probcomp** [comment]
Sorry, misread, this is not something your browser is fetching, is it?

tcpdump/wireshark?

**Comment by raxraxraxraxrax** [comment]
I just gave up and installed MathJax locally, but I can try uninstalling it later and doing that if we think it's worthwhile.
